internal/utils/nodeGroup.go

```diff
@@ -23,90 +29,178 @@ func CalculateNodeGroup(ctx context.Context, nodes v1.NodeList, config danav1alp
+	logr, _ := logr.FromContext(ctx)
 	labelSelector := labels.SelectorFromSet(labels.Set(nodegroup.LabelSelector))
 	listOptions := &client.ListOptions{
 		LabelSelector: labelSelector,
 	}
 	nodeList := v1.NodeList{}
 	if err := r.List(ctx, &nodeList, listOptions); err != nil {
-		return err, v1.ResourceList{}, v1.ResourceList{}
+		logr.Error(err, fmt.Sprintf("Error listing the nodes for the nodeGroup %s", nodegroup))
+		return err, v1.ResourceList{}
 	}
 	nodeResources := CalculateNodeGroup(ctx, nodeList, *config, nodegroup.Name)
-	groupReserved := CaculateGroupReservedResources(config.Status.ReservedResources, nodegroup.Name, config.Spec.ReservedHoursTolive)
-	totalResources := MergeTwoResourceList(nodeResources, groupReserved)
-	resourcesDiff := SubstractTwoResourceList(totalResources, sns.Spec.ResourceQuotaSpec.Hard)
-	plus, debt := GetPlusAndDebtResourceList(resourcesDiff)
-	return nil, plus, debt
+	return nil, nodeResources
 }
 
-func IsReservedResourceExist(config danav1alpha1.NodeQuotaConfig, debt v1.ResourceList, nodeGroupName string) bool {
+// doesReservedResourceExist checks if a reserved resource exists in the NodeQuotaConfig for the given node group name.
+// It takes the NodeQuotaConfig and the node group name to check.
+// It returns a boolean value indicating whether the reserved resource exists or not.
+func doesReservedResourceExist(config danav1alpha1.NodeQuotaConfig, nodeGroupName string) bool {
+	if len(config.Status.ReservedResources) == 0 {
+		return false
+	}
 	for _, reservedResources := range config.Status.ReservedResources {
-		if reservedResources.NodeGroup == nodeGroupName && reflect.DeepEqual(debt, reservedResources.Resources) {
+		if reservedResources.NodeGroup == nodeGroupName {
 			return true
 		}
 	}
 	return false
 }
 
-func AddReservedToConfig(debt v1.ResourceList, nodeGroupName string, config *danav1alpha1.NodeQuotaConfig) {
-	if IsReservedResourceExist(*config, debt, nodeGroupName) {
-		config.Status.ReservedResources = append(config.Status.ReservedResources, danav1alpha1.ReservedResources{
-			NodeGroup: nodeGroupName,
-			Resources: debt,
-			Timestamp: metav1.Now(),
-		})
+// UpdateRootSubnamespace updates the resourceQuota of the rootSubnamespace with the new quantity of resources.
+func UpdateRootSubnamespace(ctx context.Context, rootResources v1.ResourceList, rootSubnamespace danav1alpha1.SubnamespacesRoots, logger logr.Logger, client client.Client) error {
+	rootRQ, err := GetRootQuota(client, ctx, rootSubnamespace.RootNamespace)
+	if err != nil {
+		logger.Error(err, fmt.Sprintf("Error getting the %s resourceQuota", rootSubnamespace.RootNamespace))
+		return err
+	}
+	rootRQ.Spec.Hard = rootResources
+	logger.Info(fmt.Sprintf("Updating RootSubnamespace %s with new resources", rootSubnamespace.RootNamespace))
+	if err := client.Update(ctx, &rootRQ); err != nil {
+		logger.Error(err, fmt.Sprintf("Error updating rootSubnamespace %s", rootSubnamespace.RootNamespace))
+		return err
+	}
+	return nil
+}
```

Comment:
The update here is problematic since it overrides quota values which are not controlled by the `NodeQuotaConfig` and can cause the update of the secondary roots to later fail